_tutorials/2016-09-08-grpc-in-python-for-ios-xr.md

```diff
@@ -0,0 +1,210 @@
+---
+published: true
+date: '2016-09-08 09:18 -0700'
+title: gRPC in Python for IOS-XR
+author: Karthik Kumaravel
+excerpt: gRPC in Python for IOS-XR
+---
+{% include toc icon="table" title="IOS-XR: gRPC in Python" %}
+
+## Introduction
+The goal of this tutorial is to set up gRPC in Python to send gRPC commands to an IOS-XR box. This tutorial assumes that you have gone through the XR Toolbox Series before. If you haven't checked out the earlier parts to the XR toolbox Series, then you can do so here:  
+
+>
+[XR Toolbox Series]({{ base_path }}/tags/#xr-toolbox)
+
+## Pre-requisites
+
+Before we begin, let's make sure you've set up your development environment.
+If you haven't checked it out, go through the "App-Development Topology" tutorial here:  
+
+>
+[XR Toolbox, Part 3: App Development Topology]({{ base_path }}/tutorials/2016-06-06-xr-toolbox-app-development-topology)  
+  
+Follow the instructions to get your topology up and running as shown below:  
+
+![app dev topo](https://xrdocs.github.io/xrdocs-images/assets/tutorial-images/app_dev_topology.png)
+
+
+If you've reached the end of the above tutorial, you should be able to issue `vagrant status` in the `vagrant-xrdocs/lxc-app-topo-bootstrap` directory to see a rtr (IOS-XR) and a devbox (Ubuntu/trusty) instance running.  
+
+**Ensure you have the latest version of IOS-XRv.**
+{: .notice--danger}
+
+## Installing gRPC on the devbox 
+
+Let's start by installing a few developer tools on the devbox.
+
+```shell
+sudo apt-get update
+sudo apt-get -y install python-dev python-pip git
+```
+
+Now that we have installed the developer tools, lets install gRPC for Python on the box.
+
+```shell
+sudo pip install grpcio
+```
+
+Thats it! gRPC for Python is installed on the devbox.
+{: .notice--success}  
+
+## Cloning the git repo
+
+Now that we have gRPC for Python installed on the devbox. We need to get the bindings associated with IOS-XR. Let's use the library that has these bindings done.
+
+Clone the gRPC for Python library here: <https://github.com/skkumaravel/iosxr-ez/tree/master/grpc>
+
+```shell
+cd ~/
+git clone https://github.com/cisco-grpc-connection-libs/ios-xr-grpc-python.git
+cd ios-xr-grpc-python
+```
+
+## Understanding the topology
+
```

Comment:
For this topology I was using a bootstrapped topology that you have previously used, rather than creating a new repository. The default port for the IOS-XRv is always 57777 and I don't change it. I just wanted them to be able to see where they can see the port. I had a few people ask me this question when they are trying with physical boxes so I thought to put this detail part in. If you think its better to create a new repository in xrdocs I could. Let me know.